c++/src/kj/string.c++

```diff
@@ -33,6 +34,71 @@ namespace kj {
 // Warns that sprintf() is buffer-overrunny. We know that, it's cool.
 #endif
 
+namespace {
+
+long long parseSigned(const StringPtr& s, long long min, long long max) {
+  KJ_REQUIRE(s != nullptr, "String does not contain valid number", s);
+  char *endPtr;
+  errno = 0;
+  auto value = strtoll(s.begin(), &endPtr, 10);
+  KJ_REQUIRE(endPtr == s.end(), "String does not contain valid number", s);
+  KJ_REQUIRE(errno != ERANGE, "Value out-of-range", s);
+  KJ_REQUIRE(value >= min && value <= max, "Value out-of-range", value, min, max);
+  return value;
+}
+
+unsigned long long parseUnsigned(const StringPtr& s, unsigned long long max) {
+  KJ_REQUIRE(s != nullptr, "String does not contain valid number", s);
+  char *endPtr;
+  errno = 0;
+  auto value = strtoull(s.begin(), &endPtr, 10);
+  KJ_REQUIRE(endPtr == s.end(), "String does not contain valid number", s);
+  KJ_REQUIRE(errno != ERANGE, "Value out-of-range", s);
+  KJ_REQUIRE(value <= max, "Value out-of-range", value, max);
+  KJ_REQUIRE(s[0] != '-', "Value out-of-range", s); //strtoull("-1") does not fail with ERANGE
+  return value;
+}
+
+template <typename T>
+T parseInteger(const StringPtr& s) {
+  if (std::numeric_limits<T>::is_signed) {
+    auto min = (long long) std::numeric_limits<T>::min();
```

Comment:
You could avoid pulling in `std::numeric_limits` here by using `minValue` and `maxValue`.

    if (static_cast<T>(minValue) < 0) {
      long long min = static_cast<T>(minValue);
      long long max = static_cast<T>(maxValue);